### DISCUSSION TITLE
Does find method set up shallow tracking?

### DISCUSSION THREAD

**Discussion by nightire** [LegendApp/legend-state#284] [Q&A] [answered]
According to the document https://legendapp.com/open-source/state/usage/observable/#arrays, `find` should set up shallow tracking automatically, which means only adding or removing items cause an update. However, it doesn't behave as above in my case:

```ts
const state = observable({
  messages: [{ value: "hello" }]
})


observe(() => {
	const emptyMessage = room.messages.find((message) => message.value.peek() === "")
    if (emptyMessage) {
      console.log("Empty message found")
    }
})

state.messages.push({ value: "" }) // "Empty message found" <- this is expected
state.messages[0].value.set("") // "Empty message found" <- this is not expected
```

To achieve what I expected, I need to rewrite it as follows:

```ts
observe(() => {
    const emptyMessage = room.messages.get(true).find((message) => message.value === "")
    if (emptyMessage) {
      console.log("Empty message found")
    }
})

state.messages.push({ value: "" }) // "Empty message found"
state.messages[0].value.set("") // (nothing logged)
```

Is the document incorrect, or did I misunderstand the meaning of "shallow tracking"?

**Comment by jmeistrich** [comment, marked as answer]
Sorry for the delay! This should be fixed in the latest @next version and will be in the upcoming v3.
